src/librustc/error_codes.rs

```diff
@@ -2001,6 +2001,24 @@ a (non-transparent) struct containing a single float, while `Grams` is a
 transparent wrapper around a float. This can make a difference for the ABI.
 "##,
 
+E0697: r##"
+A closure has been used as `static`.
+
+Erroneous code example:
+
+```compile_fail,E0697
+fn main() {
+    static || {}; // used as `static`
+}
+```
+
+Closures cannot be used as `static`. They "save" the environment.
+Therefore, having a static closure with a static environment doesn't
+really make sense since all you can capture inside it would be variables
+with static lifetime. In this condition, better use a function directly.
+The easiest fix is to remove `static` keyword.
```

Comment:
While this does read okay, it might be slightly better worded as:

```
Closures cannot be used as `static`. They "save" the environment,
and as such a static closure would save only a static environment
which would consist only of variables with a static lifetime.  Given
this it would be better to use a proper function.  The easiest fix
is to remove the `static` keyword.
```

If you wanted a smaller change than simply replace "In this condition, better use a function directly" with "Given this limitation it would be better to use a function directly."

Reply:
thanks! applied your suggestion.